pkg/scheduler/util/utils_test.go

```diff
@@ -122,54 +126,96 @@ func TestMoreImportantPod(t *testing.T) {
+					ObjectMeta: metav1.ObjectMeta{Name: "foo1"},
+					Status:     v1.PodStatus{NominatedNodeName: "node1"},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo2"},
+					Status:     v1.PodStatus{NominatedNodeName: "node2"},
+				},
+			},
+			expectedPatchRequests: 2,
+			expectedPatchData:     []string{`{"status":{"nominatedNodeName":null}}`, `{"status":{"nominatedNodeName":null}}`},
+		},
+		{
+			name: "Should make patch request only for pods that have NominatedNodeName",
+			pods: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo1"},
+					Status:     v1.PodStatus{NominatedNodeName: ""},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo2"},
+					Status:     v1.PodStatus{NominatedNodeName: "node1"},
+				},
+			},
+			expectedPatchRequests: 1,
+			expectedPatchData:     []string{`{"status":{"nominatedNodeName":null}}`},
 		},
 		{
-			name:                     "Should not make patch request if nominated node is already cleared",
-			currentNominatedNodeName: "",
-			expectedPatchRequests:    0,
+			name: "Should make patch requests for all pods even if a patch for one pods fails",
+			pods: []*v1.Pod{
+				{
+					// In this test, the pod named "err" returns an error in the patch.
+					ObjectMeta: metav1.ObjectMeta{Name: "err"},
+					Status:     v1.PodStatus{NominatedNodeName: "node1"},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo1"},
+					Status:     v1.PodStatus{NominatedNodeName: "node1"},
+				},
+			},
+			patchError:            statusErr,
+			expectedPatchError:    utilerrors.NewAggregate([]error{statusErr}),
+			expectedPatchRequests: 2,
+			expectedPatchData:     []string{`{"status":{"nominatedNodeName":null}}`},
 		},
 	}
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			actualPatchRequests := 0
-			var actualPatchData string
+			var actualPatchData []string
 			cs := &clientsetfake.Clientset{}
 			cs.AddReactor("patch", "pods", func(action clienttesting.Action) (bool, runtime.Object, error) {
 				actualPatchRequests++
 				patch := action.(clienttesting.PatchAction)
-				actualPatchData = string(patch.GetPatch())
+				// If the pod name is "err", return an error.
+				if patch.GetName() == "err" {
+					return true, nil, test.patchError
+				}
+				actualPatchData = append(actualPatchData, string(patch.GetPatch()))
 				// For this test, we don't care about the result of the patched pod, just that we got the expected
 				// patch request, so just returning &v1.Pod{} here is OK because scheduler doesn't use the response.
 				return true, &v1.Pod{}, nil
-			})
 
-			pod := &v1.Pod{
-				ObjectMeta: metav1.ObjectMeta{Name: "foo"},
-				Status:     v1.PodStatus{NominatedNodeName: test.currentNominatedNodeName},
-			}
+			})
 
-			if err := ClearNominatedNodeName(cs, pod); err != nil {
-				t.Fatalf("Error calling removeNominatedNodeName: %v", err)
+			if err := ClearNominatedNodeName(cs, test.pods...); err != nil && err.Is(test.expectedPatchError) {
```

Comment:
Compare errors by `!reflect.DeepEqual(err.Errors(), test.expectedPatchError.Errors())`. (Your approach may miss some wrong test cases.)

Reply:
> avoid using reflect.DeepEqual with errors deepequalerrors

yes, I tried to do it first. But I got the above warning in my code editor. No problem? (The test succeeds.)

Reply:
What kind of error? I don't see a problem with this approach of comparison.

Reply:
I don't see a problem with this method of comparison.

Reply:
Not error but warning. And warning text is just this.
> avoid using reflect.DeepEqual with errors 

It is displayed only on the code editor (I use vscode) not on the terminal. So, I don't think it's a problem.